tests/rptest/services/utils.py

```diff
@@ -6,20 +6,40 @@
 from typing import Generator, Optional
 
 from rptest.clients.kubectl import KubectlTool, KubeNodeShell
-from rptest.services.cloud_broker import CloudBroker
+
+
+class Stopwatch():
```

Comment:
Semantics of reset/start/stop are not very clear.

If I do:

```
s = Stopwatch()
# a
s.stop()
# c
s.start()
# c
e.stop()
```

Which intervals should be included? It seems to me like the current implementation will return a negative elapsed time in this case.

I think a reasonable semantic is that all intervals between stop/start are measured, which requires an elapsed accumulator in the state as well. Or another option is just the "last" interval (since the last start() if stop() has not been called since or the stop() - start() interval if it has). Then you don't need `reset()`.

Reply:
Makes sense. Will update. My original goal was to simulate actual stopwatch functionality when only one single interval is measured; a.k.a. last one, like on a real watch.